drivers/codec/wm8960.py

```diff
@@ -0,0 +1,783 @@
+_POWER3_ROMIX_MASK = const(0x04)
+
+# Register _DACCTL1 .. 3
+_DACCTL1_MONOMIX_MASK = const(0x10)
+_DACCTL1_MONOMIX_SHIFT = const(0x4)
+_DACCTL1_DACMU_MASK = const(0x08)
+_DACCTL1_DEEM_MASK = const(0x06)
+_DACCTL1_DEEM_SHIFT = const(0x01)
+_DACCTL2_DACSMM_MASK = const(0x08)
+_DACCTL2_DACMR_MASK = const(0x04)
+_DACCTL3_ALCSR_MASK = const(0x07)
+
+# _WM8060_ALC1 .. 3
+_ALC_CHANNEL_MASK = const(0x180)
+_ALC_CHANNEL_SHIFT = const(0x7)
+_ALC_MODE_MASK = const(0x100)
+_ALC_MODE_SHIFT = const(0x8)
+_ALC_GAIN_MASK = const(0x70)
+_ALC_GAIN_SHIFT = const(0x4)
+_ALC_TARGET_MASK = const(0x0F)
+_ALC_ATTACK_MASK = const(0x0F)
+_ALC_DECAY_MASK = const(0xF0)
+_ALC_DECAY_SHIFT = const(4)
+_ALC_HOLD_MASK = const(0xF)
+
+# Register _NOISEG
+_NOISEG_LEVEL_SHIFT = const(3)
+
+_I2C_ADDR = const(0x1A)
+
+# WM8960 maximum volume values
+_MAX_VOLUME_ADC = const(0xFF)
+_MAX_VOLUME_DAC = const(0xFF)
+_MAX_VOLUME_HEADPHONE = const(0x7F)
+_MAX_VOLUME_LINEIN = const(0x3F)
+_MAX_VOLUME_SPEAKER = const(0x7F)
+
+# Config symbol names
+# Modules
+module_ADC = const(0)  # ADC module in WM8960
+module_DAC = const(1)  # DAC module in WM8960
+module_VREF = const(2)  # VREF module
+module_headphone = const(3)  # Headphone
+module_mic_bias = const(4)  # Mic bias
+module_mic = const(5)  # Input Mic
+module_line_in = const(6)  # Analog in PGA
+module_line_out = const(7)  # Line out module
+module_speaker = const(8)  # Speaker module
+module_omix = const(9)  # Output mixer
+module_mono_out = const(10)  # Mono mix
+
+# Route
+route_bypass = const(0)  # LINEIN->Headphone.
+route_playback = const(1)  #  I2SIN->DAC->Headphone.
+route_playback_record = const(2)  # I2SIN->DAC->Headphone, LINEIN->ADC->I2SOUT.
+route_record = const(5)  # LINEIN->ADC->I2SOUT.
+
+# Input
+input_closed = const(0)  # Input device is closed
+input_mic1 = const(1)  # Input as single ended mic, only use L/RINPUT1
+input_mic2 = const(2)  # Input as diff. mic, use L/RINPUT1 and L/RINPUT2
+input_mic3 = const(3)  # Input as diff. mic, use L/RINPUT1 and L/RINPUT3
+input_line2 = const(4)  # Input as line input, only use L/RINPUT2
+input_line3 = const(5)  # Input as line input, only use L/RINPUT3
+
+# ADC sync input
+sync_adc = const(0)  # Use ADCLRC pin for ADC sync
+sync_dac = const(1)  # used DACLRC pin for ADC sync
+
+# Protocol type
+bus_I2S = const(2)  # I2S type
+bus_left_justified = const(1)  # Left justified mode
+bus_right_justified = const(0)  # Right justified mode
+bus_PCMA = const(3)  # PCM A mode
+bus_PCMB = const(3 | (1 << 4))  # PCM B mode
+
+# Channel swap
+swap_none = const(0)
+swap_input = const(1)
+swap_output = const(2)
+
+# Mute settings
+mute_fast = const(0)
+mute_slow = const(1)
+
+# ALC settings
+alc_off = const(0)
+alc_right = const(1)
+alc_left = const(2)
+alc_stereo = const(3)
+alc_mode = const(0)  # ALC mode
+alc_limiter = const(1)  # Limiter mode
+
+# Clock Source
+sysclk_mclk = const(0)  # sysclk source from external MCLK
+sysclk_PLL = const(1)  # sysclk source from internal PLL
+
+
+# fmt: off
```

Comment:
Please move this `fmt: off` down below to just before the `cache = ...` line.

Reply:
Done